### DISCUSSION TITLE
CrossOver bottle "Heroic" does not exist - can't launch (macOS)

### DISCUSSION THREAD

**Discussion by IanDavies** [Heroic-Games-Launcher/HeroicGamesLauncher#1567] [Q&A]
I have CrossOver and it works just fine (macOS Big Sur 11.6.7) and I'm trying to get Heroic to work with it. Heroic recognises CrossOver as the Wine version, however despite creating a bottle named "Heroic" as per instructions, Heroic reports that it doesn't exist and can't launch: [Image link](https://i.imgur.com/fo7398Q.png)

I've tried various bottle environments (Win 7 64-bit, Win 10 64-bit etc.) but nothing seems to make any difference.
The only slightly non-standard thing is that I have my CrossOver bottle directory on an external drive rather than the default location, but I don't see anywhere in the Heroic interface to specify that. I assumed if CrossOver knew where the bottles were, Heroic would too?

Any suggestions?

**Comment by flavioislima** [comment]
Being on an external drive should not be an issue.
Try renaming the bottle to Legendary and run an Epic game afterward.

@CommandMC I remember you did some refactoring to check the bottle's existence. Maybe you know better what's happening.

**Reply by CommandMC** [reply]
Heroic checks if `~/Library/Application Support/CrossOver/Bottles/Heroic/cxbottle.conf` exists to determine that the bottle is valid. I've tested this on a Mac VM and it worked fine there, and the same check is also done by Legendary (in fact I got that path from its code)

@IanDavies Could you check if the file/folder in question exists?

I assume CX doesn't just symlink the `Bottles` directory, and instead stores the path in a configuration file somewhere

**Reply by IanDavies** [reply]
Thanks both of you for looking at this.

So `~/Library/Application Support/CrossOver/Bottles/Heroic/cxbottle.conf` does exist, however, it's left over from before I changed the bottle directory to an external drive, and all the last modified dates confirm it's not the directory CrossOver is using any more. I followed [this guide](https://www.codeweavers.com/support/wiki/mac/faq/cxofficemacbottlelocation) from Codeweavers, using the `defaults write` method rather than directly editing the .conf file.

The Crossover directory I changed it to looks like this: ![image](https://i.imgur.com/tgqvTcN.png)
It doesn't look like CrossOver creates a `Bottles` directory by default, just keeps them in the top level with everything else it creates, but it works.

**Reply by CommandMC** [reply]
Hm, well, that `Bottles` directory is what both Heroic and Legendary check for (so even if we would check for the right one, Legendary would still complain)

Could you try running `ln -s /path/to/your/new/bottles/dir "~/Library/Application Support/CrossOver/Bottles"` to symlink the old directory to the new one?

**Reply by acryllect** [reply]
I'll chime in to revive this thread, I have this same issue. Tried using a symlink to my new bottle directory but it still doesn't recognize the "Heroic" bottle.

**Reply by tonicastillo** [reply]
The symlink worked for me. thanks

**Reply by adriengonz** [reply]
Same issue here with my bottles in external drive, the symlink worked for me too

**Reply by DashingDave** [reply]
I have 5 bottles in a folder called Bottles on my external drive named Red. Using the nice capability of just dragging an item from the Finder into a Terminal window to enter the path, the command I entered was

`home@myStudio ~ % ln -s /Volumes/Red/CrossOver/Bottles /Users/home/Library/Application\ Support/CrossOver/Bottles
`

[No 'reply' came back, just a new line 'in' the home folder (myStudio is the name of my Mac, in case that's not clear).]

However, when opening CrossOver—with the Bottles location (already) set to the default folder—the bottles don't appear in CrossOver's sidebar. Does this method not work anymore? Or is it working for some people and not for others?

**Reply by akshin-autods** [reply]
this one worked for me : 
1. Prepare the destination
First, we must make sure the "fake" location is empty so we can put the link there.

Open Terminal (Applications > Utilities > Terminal).
Copy and paste this command and press Enter:
`rm -rf ~/Library/Application\ Support/CrossOver/Bottles`
(This removes the empty or broken "Bottles" folder in your Library so we can replace it with the link.)
2. Create the Symlink
Now, run the link command. Note: If your bottles are on an external drive, replace /Volumes/YourDriveName/Bottles with your actual path.

If your bottles are in the standard location but Heroic is blind to them, run this:
`ln -s ~/Library/Application\ Support/CrossOver/Bottles ~/.local/share/wineprefixes`

then i created new bottle , and restarted heroic
